api/commands/run.ts

```diff
@@ -112,24 +112,54 @@ export class Runner {
   }
 
   private async executeAllActionsReadyForExecution() {
-    const allSuccessfulActions = this.result.actions.filter(isSuccessfulAction).map(fn => fn.name);
-    const isReadyForExecution = (action: dataform.IExecutionAction) => {
-      for (const dependency of action.dependencies) {
-        if (!allSuccessfulActions.includes(dependency)) {
-          return false;
-        }
-      }
-      return true;
-    };
-    const readyForExecutionActions = this.pendingActions.filter(isReadyForExecution);
-    this.pendingActions = this.pendingActions.filter(action => !isReadyForExecution(action));
-    return Promise.all(
-      readyForExecutionActions.map(async action => {
-        this.result.actions.push(await this.executeAction(action));
-        await this.triggerChange();
-        await this.executeAllActionsReadyForExecution();
-      })
+    // If the run has been cancelled, skip all pending actions.
+    if (this.cancelled) {
+      const allPendingActions = this.pendingActions;
+      this.pendingActions = [];
+      await Promise.all(
+        allPendingActions.map(async pendingAction => this.skipAction(pendingAction))
```

Comment:
These aren't promises, no need to await or promise.all() here. `skipAction` implies it mutates state, but it does not - `skippedAction` or `getSkippedAction` ? Or could just be inline tbh...

Reply:
fixed, i had accidentally not called `onActionExecutionComplete` on these (which does return a Promise)